src/savegame_json.cpp

```diff
@@ -2528,20 +2529,22 @@ void map_memory::store( JsonOut &jsout ) const
 void map_memory::load( JsonObject &jsin )
 {
     JsonArray map_memory_tiles = jsin.get_array( "map_memory_tiles" );
-    memorized_terrain.clear();
+    tiles.clear();
+    tile_map.clear();
     while( map_memory_tiles.has_more() ) {
         JsonObject pmap = map_memory_tiles.next_object();
         const tripoint p( pmap.get_int( "x" ), pmap.get_int( "y" ), pmap.get_int( "z" ) );
-        const memorized_terrain_tile m{ pmap.get_string( "tile" ), pmap.get_int( "subtile" ), pmap.get_int( "rotation" ) };
-        memorized_terrain[p] = m;
+        memorize_tile( std::numeric_limits<int>::max(), p, pmap.get_string( "tile" ),
+                       pmap.get_int( "subtile" ), pmap.get_int( "rotation" ) );
```

Comment:
Invoking `memorize_tile` seems to be more expensive than populating the data directly. I don't know if it really slows down loading even a bit, but it might in case of a huge number of tiles/symbols.

Reply:
I might extract a private helper that bypasses the limit for one thing, otherwise I was just trying to keep it DRY.